File: src/main/java/com/enzo/testemuralis/models/dto/EnderecoResponseDTO.java
package com.enzo.testemuralis.models.dto;

import com.enzo.testemuralis.models.Endereco;

public record EnderecoResponseDTO(
        String cep,
        String logradouro,
        String cidade,
        String numero,
        String complemento) {
    public EnderecoResponseDTO(Endereco endereco) {
        this(
                endereco.getCep(),
                endereco.getLogradouro(),
                endereco.getCidade(),
                endereco.getNumero(),
                endereco.getComplemento());
    }
}

File: dotnet/TesteMuralis/Dto/EnderecoResponseDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TesteMuralis.Models;

namespace TesteMuralis.Dto
{
    public class EnderecoResponseDTO
    {
        public string Cep { get; set; }
        public string Logradouro { get; set; }
        public string Cidade { get; set; }
        public string Numero { get; set; }
        public string Complemento { get; set; }

        public EnderecoResponseDTO(Endereco endereco)
        {
            Cep = endereco.Cep;
            Logradouro = endereco.Logradouro;
            Cidade = endereco.Cidade;
            Numero = endereco.Numero;
            Complemento = endereco.Complemento;
        }
    }
}